lib/graphiti/resource/persistence.rb

```diff
@@ -69,74 +69,82 @@ def add_callback(kind, lifecycle, method = nil, only, &blk)
 
-      def create(create_params)
+      def create(create_params, meta = nil)
         model_instance = nil
 
-        run_callbacks :persistence, :create, create_params do
-          run_callbacks :attributes, :create, create_params do |params|
-            model_instance = build(model)
-            assign_attributes(model_instance, params)
+        run_callbacks :persistence, :create, create_params, meta do
+          run_callbacks :attributes, :create, create_params, meta do |params|
+            model_instance = call_method(:build, model, meta)
+            call_method(:assign_attributes, model_instance, params, meta)
             model_instance
           end
 
-          run_callbacks :save, :create, model_instance do
-            model_instance = save(model_instance)
+          run_callbacks :save, :create, model_instance, meta do
+            model_instance = call_method(:save, model_instance, meta)
           end
 
           model_instance
         end
       end
 
-      def update(update_params)
+      def update(update_params, meta = nil)
         model_instance = nil
         id = update_params.delete(:id)
 
-        run_callbacks :persistence, :update, update_params do
-          run_callbacks :attributes, :update, update_params do |params|
+        run_callbacks :persistence, :update, update_params, meta do
+          run_callbacks :attributes, :update, update_params, meta do |params|
             model_instance = self.class._find(params.merge(id: id)).data
-            assign_attributes(model_instance, params)
+            call_method(:assign_attributes, model_instance, params, meta)
             model_instance
           end
 
-          run_callbacks :save, :update, model_instance do
-            model_instance = save(model_instance)
+          run_callbacks :save, :update, model_instance, meta do
+            model_instance = call_method(:save, model_instance, meta)
           end
         end
 
         model_instance
       end
 
-      def destroy(id)
+      def destroy(id, meta = nil)
         model_instance = self.class._find(id: id).data
-        run_callbacks :destroy, :destroy, model_instance do
-          delete(model_instance)
+        run_callbacks :destroy, :destroy, model_instance, meta do
+          call_method(:delete, model_instance, meta)
         end
         model_instance
       end
 
-      def build(model)
+      def build(model, meta = nil)
         adapter.build(model)
       end
 
-      def assign_attributes(model_instance, update_params)
+      def assign_attributes(model_instance, update_params, meta = nil)
         adapter.assign_attributes(model_instance, update_params)
       end
 
-      def save(model_instance)
+      def save(model_instance, meta = nil)
         adapter.save(model_instance)
       end
 
-      def delete(model_instance)
+      def delete(model_instance, meta = nil)
         adapter.destroy(model_instance)
       end
 
       private
 
+      def call_method(meth, *args)
```

Comment:
Assuming this is for backwards-compatibility purposes?  Since we haven't released 1.0 yet I'd be in favor of making this a breaking change to avoid a long term support issue. At the very least we could add a deprecation warning if it's the legacy arity, saying that it will go away and give a bit to upgrade.

Additionally, I'd make this method name more descriptive of the purpose since it's specific to handling the `meta` argument.

Reply:
It's more because >80% of the time you don't care about `meta`. IOW, you should not even have to know about `meta` and do:

```ruby
def save(model)
  Rails.logger.info "saving!"
  super
end
```

And not blow up on accepting one argument instead of two.

Agree on descriptive name though.